oauth2_http/java/com/google/auth/oauth2/TrustBoundary.java

```diff
@@ -0,0 +1,192 @@
+ * distribution.
+ *
+ *    * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.auth.oauth2;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpBackOffIOExceptionHandler;
+import com.google.api.client.http.HttpBackOffUnsuccessfulResponseHandler;
+import com.google.api.client.http.HttpIOExceptionHandler;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpUnsuccessfulResponseHandler;
+import com.google.api.client.json.GenericJson;
+import com.google.api.client.json.JsonParser;
+import com.google.api.client.util.ExponentialBackOff;
+import com.google.api.client.util.Key;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.common.base.MoreObjects;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a trust boundary that can be used to restrict access to resources. This is an
+ * experimental feature.
+ */
+public final class TrustBoundary {
+
+  static final String TRUST_BOUNDARY_KEY = "x-allowed-locations";
+  static final String GOOGLE_AUTH_TRUST_BOUNDARY_ENABLED_ENV_VAR =
+      "GOOGLE_AUTH_TRUST_BOUNDARY_ENABLE_EXPERIMENT";
+  private static final String NO_OP_VALUE = "0x0";
+  private final String encodedLocations;
+  private final List<String> locations;
+
+  public TrustBoundary(String encodedLocations, List<String> locations) {
+    this.encodedLocations = encodedLocations;
+    this.locations =
+        locations == null
+            ? Collections.<String>emptyList()
+            : Collections.unmodifiableList(locations);
+  }
+
+  private static EnvironmentProvider environmentProvider = SystemEnvironmentProvider.getInstance();
+
+  public String getEncodedLocations() {
+    return encodedLocations;
+  }
+
+  public List<String> getLocations() {
+    return locations;
+  }
```

Comment:
For all other public methods, can you add public javadocs?

Not required for package-private, but definitely would help with future maintaince! Thanks!